### PR TITLE
Dto objects prototyped

### PR DESCRIPTION
**Pull request by Michael-Kochis**
Philip,

This is the first version of the DTO objects, per your guidelines.

### CODE REVIEW

File: src/main/java/com/revature/g2g/models/SkillDTO.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	private long skillId;
+	private String name;
+	private SkillDTO parentSkill;
+
+	public SkillDTO(Skill skill) {
+		super();
+		this.skillId = skill.getSkillId();
+		this.name = skill.getName();
+		this.parentSkill = new SkillDTO(skill.getParentSkill() );
+	}
```
Comment (pjlawrence) [inline src/main/java/com/revature/g2g/models/SkillDTO.java]: We don't want this to infinitely loop if they create a loop of skills. Maybe overload with a limiter?
`public SkillDTO(Skill skill, boolean stopParentSkill){
    super();
    this.skillId = skill.getSkillId();
    this.name = skill.getName();
    if (stopParentSkill){
        this.parentSkill = new SkillDTO(skill.getParentSkill(), false);
    } else{
        this.parentSkill = null;
    }
}`

File: src/main/java/com/revature/g2g/services/jda/listeners/GuildVoiceEventListener.java [hunk 2]
```diff
@@ -91,7 +91,6 @@ private void processMoveArrive(GuildVoiceUpdateEvent event) {
 		}
 	}
 	private void processMoveLeave(GuildVoiceUpdateEvent event){
-		//TODO add logic to mark person as having left.
```
Comment (pjlawrence) [inline src/main/java/com/revature/g2g/services/jda/listeners/GuildVoiceEventListener.java:94]: I don't see new code resolving this todo. If it is not solved, please leave the to do for a future sprint. (this specific todo is dependant on OAuth working so it is currently blocked)

**Review by IncarnateGaming — CHANGES_REQUESTED** [review]

File: src/main/java/com/revature/g2g/models/RoomDTO.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.revature.g2g.models;
+
+import java.util.Date;
+
+public class RoomDTO {
```
Comment (pjlawrence) [inline src/main/java/com/revature/g2g/models/RoomDTO.java]: lombock annotations

File: src/main/java/com/revature/g2g/models/SkillDTO.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.revature.g2g.models;
+
+public class SkillDTO {
```
Comment (pjlawrence) [inline src/main/java/com/revature/g2g/models/SkillDTO.java]: Lombok annotations

**Review by IncarnateGaming — CHANGES_REQUESTED** [review]

**Pull request merged**